crates/lib/src/cli.rs

```diff
@@ -763,6 +763,51 @@ fn has_soft_reboot_capability(deployment: Option<&crate::spec::BootEntry>) -> bo
     deployment.map(|d| d.soft_reboot_capable).unwrap_or(false)
 }
 
+/// Check if SELinux policy differs between booted and target deployments.
+/// Returns an error if SELinux is enabled and the policies differ.
+#[context("Checking SELinux policy compatibility with soft reboot")]
+fn check_selinux_policy_for_soft_reboot(
```

Comment:
This helps, but I think what we really want is to change `has_soft_reboot_capability`.
 
But then that raises the next thing in that in theory this change should go in libostree which has a `deployment_can_soft_reboot` API which already does checks like this.

OTOH, I don't mind filling it in here either, we can just have a `// TODO lower into ostree` to start.

The other thing of course related to this is that when we get around to soft reboots + composefs, we'll need similar logic there.